### PR TITLE
Added "expanded" getter/setter to API. Refactored focus management.

### PR DESCRIPTION
**Pull request by ianmcburnie**
This PR has a bunch of cleanup related to focus management, event handlers and the API. Plus a few other misc things. It's all backwards compatible, but I might bump the minor version given the scale of changes.

For the reviewer I would recommend looking at the end code rather than the diff. It should be a little easier to follow than it was previously.

SQUASH: YES PLEASE.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,10 +68,48 @@ Clicking the button will now toggle it's aria-expanded state. CSS can be used to
...
+    display: block;
+}
+```
+
 ## Params
 
 * `el`: the root widget el
-* `options.alwaysDoFocusManagement`: honour `focusManagement` setting even when non-keyboard device is used (default: false)
+* `options.alwaysDoFocusManagement`: whether `focusManagement` option (see below) should apply for mouse click
+* `options.ariaControls`: specify whether `aria-controls` relationship should be created between host and overlay (default: true)
```
Comment (seangates) [inline README.md]: Why would there not be a relationship when using the expander?

**Review by seangates — COMMENTED** [review]
This seems much better! Just have that one question.

File: README.md (hunk 1 above)
Comment (ianmcburnie) [reply README.md]: Depending on the type of widget (menu, fake-menu, listbox, combobox, infotip, etc), aria-controls isn't always needed. In fact, some a11y checkers (axe, for example) are flagging it as an error. So I may need to remove aria-controls in one or two of our components. I need to dig into that a little more when I get time.

**Comment by ianmcburnie** [comment]
@seangates I appreciate the review on this. I have one more for next-id: https://github.com/makeup-js/makeup-next-id/pull/1. It's not urgent, so no hurry.

**Review by seangates — APPROVED** [review]

**Pull request merged**